### PR TITLE
[msbuild] Remove BTouchReferencePath in order to fix bug 56317

### PR DESCRIPTION
**Pull request by dalexsoto**
https://bugzilla.xamarin.com/show_bug.cgi?id=56317

BTouchReferencePath scans for satellite assemblies and brings
dlls that are not referenced from within the binding project and
this causes compiler error CS1704 when dlls that contain
localized resources and said dlls are named equally.

### CODE REVIEW

File: msbuild/Xamarin.iOS.Tasks.Core/Xamarin.iOS.ObjCBinding.CSharp.targets [hunk 1]
```diff
@@ -46,10 +46,6 @@ Copyright (C) 2013-2016 Xamarin Inc. All rights reserved.
 		Inputs="$(MSBuildAllProjects);@(ObjcBindingApiDefinition);@(ObjcBindingCoreSource);@(ReferencePath);@(ObjcBindingNativeLibrary)"
 		Outputs="$(_GeneratedSourcesFileList)">
 
-		<ItemGroup>
-			<BTouchReferencePath Include="@(ReferenceCopyLocalPaths)" Condition="'%(Extension)' == '.dll'" />
-		</ItemGroup>
```
Comment (spouliot) [inline msbuild/Xamarin.iOS.Tasks.Core/Xamarin.iOS.ObjCBinding.CSharp.targets]: This was added in maccore (before open sourcing the SDK):
```
commit f0a86283491f0267f55080e4ece49bcd65ebebc1
Author: Jeffrey Stedfast <jeff@xamarin.com>
Date:   Thu Oct 16 15:45:34 2014 -0400

    [msbuild] Attempt at fixing bug #22135
```

references:
https://bugzilla.xamarin.com/show_bug.cgi?id=22135
https://bugzilla.xamarin.com/show_bug.cgi?id=23674

so it's worth double checking the original test cases still works.

**Comment by spouliot** [comment]
c.c. @emaf since this changes an msbuild file that XVS also uses
c.c. @jstedfast and @VincentDondain to review too

File: msbuild/Xamarin.iOS.Tasks.Core/Xamarin.iOS.ObjCBinding.CSharp.targets (hunk 1 above)
Comment (dalexsoto) [inline msbuild/Xamarin.iOS.Tasks.Core/Xamarin.iOS.ObjCBinding.CSharp.targets]: Test case added in 8a6615387017a6f286b9ecacd3a9c879e961b20f

**Comment by dalexsoto** [comment]
@rolfbjarne / @spouliot test cases added

File: msbuild/tests/MySatelliteAssembliesBug/iOSBinding/ApiDefinition.cs [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+	//     -(id)initWithElmo:(ElmoMuppet *)elmo;
+	//
+	// Can be bound as:
+	//
+	//     [Export ("initWithElmo:")]
+	//     IntPtr Constructor (ElmoMuppet elmo);
+	//
+	// For more information, see http://developer.xamarin.com/guides/ios/advanced_topics/binding_objective-c/
+	//
+}
```
Comment (spouliot) [inline msbuild/tests/MySatelliteAssembliesBug/iOSBinding/ApiDefinition.cs]: is that file needed, it has no code ?

File: msbuild/tests/MySatelliteAssembliesBug/iOSBinding/Properties/AssemblyInfo.cs [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+// The form "{Major}.{Minor}.*" will automatically update the build and revision,
+// and "{Major}.{Minor}.{Build}.*" will update just the revision.
+
+[assembly: AssemblyVersion ("1.0.*")]
+
+// The following attributes are used to specify the signing key for the assembly, 
+// if desired. See the Mono documentation for more information about signing.
+
+//[assembly: AssemblyDelaySign(false)]
+//[assembly: AssemblyKeyFile("")]
```
Comment (spouliot) [inline msbuild/tests/MySatelliteAssembliesBug/iOSBinding/Properties/AssemblyInfo.cs]: file likely unneeded to make the test case

File: msbuild/tests/MySatelliteAssembliesBug/FooAssembly/Properties/AssemblyInfo.cs [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+// The form "{Major}.{Minor}.*" will automatically update the build and revision,
+// and "{Major}.{Minor}.{Build}.*" will update just the revision.
+
+[assembly: AssemblyVersion ("1.0.*")]
+
+// The following attributes are used to specify the signing key for the assembly, 
+// if desired. See the Mono documentation for more information about signing.
+
+//[assembly: AssemblyDelaySign(false)]
+//[assembly: AssemblyKeyFile("")]
```
Comment (spouliot) [inline msbuild/tests/MySatelliteAssembliesBug/FooAssembly/Properties/AssemblyInfo.cs]: file likely unneeded to create a test case

**Comment by dalexsoto** [comment]
@spouliot Fixed 👍
